src/policy.rs

```diff
@@ -107,11 +128,38 @@ impl Display for Policy {
 }
 
 impl Policy {
+    /// Converts the given string into a Policy. Does not fail if the given
+    /// string uses the legacy format.
+    pub fn parse_and_convert(string: &str) -> Result<Self, Error> {
+        match serde_json::from_str(string) {
+            Ok(policy) => Ok(policy),
+            Err(e) => {
+                if let Ok(policy) = serde_json::from_str::<LegacyPolicy>(string) {
+                    Ok(Policy {
+                        version: PolicyVersion::V1,
+                        max_attribute_creations: policy.max_attribute_creations,
+                        last_attribute_value: policy.last_attribute_value,
+                        axes: policy.axes,
+                        attributes: policy
+                            .attributes
+                            .into_iter()
+                            .map(|(name, values)| (name, (values, EncryptionHint::Classic)))
+                            .collect(),
+                    })
+                } else {
+                    Err(Error::DeserializationError(e))
+                }
```

Comment:
I think the `if let` is useless, no? Just use `?`?

Reply:
That's true, now that I have added a dedicated error variant, thanks :+1:

Reply:
Actually it is still useful because we are not returning the error from the `LegacyPolicy` deserialization but the `Policy` one.

Reply:
Not sure to understand what is the problem…

Reply:
Hum ok! Undertood :-) Maybe add a comment or rename `e` to something more specific?

Reply:
Yes you are right, even I got mistaken be reading my own code.